administrator/components/com_categories/models/fields/modal/category.php

```diff
@@ -157,11 +157,12 @@ protected function getInput()
 				. '<span class="icon-edit"></span>' . JText::_('JACTION_EDIT')
 				. '</a>';
 
+			$formToken = (new \Joomla\Cms\Session\CsrfToken(JFactory::getSession()))->getVarname();
```

Comment:
I see the line `new \Joomla\Cms\Session\CsrfToken(JFactory::getSession())` repeated all over the place. I would suggest replacing this with `JFactory::getToken()`. When we refactor JFactory out and use a container this would be replaced with something like `$container->token`. This would make more sense and would be one less hard dependency.